### ISSUE TITLE
Robust `git:diff` and `git: graph`

### ISSUE THREAD

**Issue by divmain** [divmain/GitSavvy#486]
Here are the options that I'd like to see.

- `git: log`
  - `All`
  - `Filtered by author`
    - Display list of all committers
    - When committer is selected, display commits filtered by author
  - `Filtered by commit message regexp`
    - Prompt user for regexp text
    - When text is entered, display commits filtered by commit message; `git log --grep="[Pp]rovided regexp"`
  - `Filtered by diff contents`
    - Prompt user for regexp text
    - When text is entered, display commits filtered by commit changes; `git log -G"[Pp]rovided regexp"`
- `git: log current file`
  - same options as those for `git: log`
- `git: graph`
  - `Current branch`
  - `All branches`
  - `Filtered by author`
  - `Filtered by commit message regexp`

**Comment by randy3k** [comment]
i guess we could duplicate all the options of `git: log` to `git: graph`.

**Comment by divmain** [comment]
Yes, that's what I intended, although it didn't come across clearly.  I've edited the description.

**Comment by stoivo** [comment]
👍  for unifying these three commands in what they prompt for.

**Comment by stoivo** [comment]
When we are talking about rewrite git log,  graph especially. There is many option like `--no-merges`, it would be nice to have a way to toggle these?

I have two proposals
1.
- `git: log`
  - `--no-merges (inactive)` -> would open the same prompt but change from inactive to active
  - `For current branch`
  - `For all branches`
  - `Filtered by author`
    - Display list of all committers
    - When committer is selected, display commits filtered by author
  - `Filtered by commit message regexp`
    - Prompt user for regexp text
    - When text is entered, display commits filtered by commit message; `git log --grep="[Pp]rovided regexp"`
  - `Filtered by diff contents`
    - Prompt user for regexp text
    - When text is entered, display commits filtered by commit changes; `git log -G"[Pp]rovided regexp"`
- `git: log current file`
  - same options as those for `git: log`
- `git: graph`
  - same options as those for `git: log` 

2.
 When the graph is open press `?`(ether a html view or a commad pannel) to toggle all kind of options

**Comment by asfaltboy** [comment]
I believe what is left to do here is:

- [ ] Toggle off/on merge commits
- [ ] Filtered by commit message regexp
        Prompt user for regexp text
        When text is entered, display commits filtered by commit message; git log --grep="[Pp]rovided regexp"
- [ ] Filtered by diff contents
        Prompt user for regexp text
        When text is entered, display commits filtered by commit changes; git log -G"[Pp]rovided regexp"

Somewhat relates to #1012
